pkg/controller/projects/deploykeys/controller.go

```diff
@@ -116,9 +116,9 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	lateInitializeProjectDeployKey(&cr.Spec.ForProvider, dk)
 	isLateInitialized := !cmp.Equal(currentState, &cr.Spec.ForProvider)
 
-	cr.Status.AtProvider = v1alpha1.DeployKeyObservation{
-		ID:        &dk.ID,
-		CreatedAt: &metav1.Time{Time: *dk.CreatedAt},
+	cr.Status.AtProvider = v1alpha1.DeployKeyObservation{ID: &dk.ID}
+	if dk.CreatedAt != nil {
+		cr.Status.AtProvider.CreatedAt = &metav1.Time{Time: *dk.CreatedAt}
```

Comment:
Instead of an if block you can use a helper function like https://github.com/crossplane-contrib/provider-aws/blob/master/pkg/clients/pointer.go#L123